File: chaincode/dto/transaction.dto.go
package dto

type CreateTransactionDto struct {
	Id           string  `json:"user_id"`
	IdKey        string  `json:"idkey"`
	Hash         string  `json:"hash"`
	Amount       float32 `json:"amount"`
	CurrencyType string  `json:"currency_type"`
	CreatedAt    string  `json:"created_at"`
}

File: dapp/src/dto/transaction.dto.ts
export interface PostTransactionDto {
    idkey: string
    amount: number
    currency_type: string
  }
export interface PostTransactionDbDto {
    idkey: string
    hash: string
    amount: number
    currency_type: string
    created_at: Date
  }
export interface CreateCCTransactionDto {
    id: string
    idkey: string
    hash: string
    amount: string
    currency_type: string
    created_at: string
  }
